### DISCUSSION TITLE
使用【1Panel】服务器面板 - 安装升级【青龙面板】

### DISCUSSION THREAD

**Discussion by QYG2297248353** [whyour/qinglong#2379] [Show and tell]
# 关于1Panel面板
1Panel 是新一代的 Linux 服务器运维管理面板

## 安装1Panel面板
访问[1Panel官网](https://1panel.cn/)，选择适合自己服务器系统的1Panel面板安装命令，按照1Panel面板[官方文档](https://1panel.cn/docs/installation/online_installation/)进行安装

#### RedHat/CentOS安装脚本
```shell
curl -sSL https://resource.fit2cloud.com/1panel/package/quick_start.sh -o quick_start.sh && sh quick_start.sh
```

#### Ubuntu安装脚本
```shell
curl -sSL https://resource.fit2cloud.com/1panel/package/quick_start.sh -o quick_start.sh && sudo bash quick_start.sh
```

#### Debian安装脚本
```shell
curl -sSL https://resource.fit2cloud.com/1panel/package/quick_start.sh -o quick_start.sh && bash quick_start.sh
```

#### openEuler/其他安装脚本
第一步：安装 docker
```shell
bash <(curl -sSL https://linuxmirrors.cn/docker.sh)
```
第二步：安装 1Panel
```shell
curl -sSL https://resource.fit2cloud.com/1panel/package/quick_start.sh -o quick_start.sh && sh quick_start.sh
```

## 安装第三方应用商店

[第三方应用商店](https://github.com/QYG2297248353/appstore-1panel) 由 `新疆萌森软件开发工作室` 维护。

### 一键式安装脚本

每间隔6小时更新一次第三方应用商店内容

#### GitHub 镜像源
安装自动化脚本
```shell
curl -sSL https://install.lifebus.top/auto_app_install.sh | bash
```
卸载自动化脚本
```shell
curl -sSL https://install.lifebus.top/auto_app_uninstall.sh | bash
```

#### Gitee (中国特供版)
安装自动化脚本
```shell
curl -sSL https://install.lifebus.top/auto_app_install_zh.sh | bash
```

卸载自动化脚本
```shell
curl -sSL https://install.lifebus.top/auto_app_uninstall_zh.sh | bash
```

### 升级第三方应用商店

升级后请执行 `更新应用列表`

#### GitHub 镜像源
```shell
curl -sSL https://install.lifebus.top/app_install.sh | bash
```

#### Gitee (中国特供版)
```shell
curl -sSL https://install.lifebus.top/app_install_zh.sh | bash
```

## 安装 青龙面板
打开 `应用商店` 模块。

### 更新应用商店列表

> 1. 勾选 `显示本地应用`
> 2. 执行 `更新应用列表`
> 3. 搜索 `青龙`

![image](https://github.com/whyour/qinglong/assets/56836951/2a2e5107-f764-482b-8091-fde85578ce2f)

> 安装时选择带有 `本地` 标签的版本，区别在于，其一：维护方不一样，其二，数据持久化方式不一样

## 对比图
+ 1Panel 官方维护版
![image](https://github.com/whyour/qinglong/assets/56836951/4e716240-135b-41f3-990f-8a93a8f8cd34)

+ 1Panel 第三方维护版
![image](https://github.com/whyour/qinglong/assets/56836951/2f27f017-9e2b-4ff2-b984-aaece38abb6b)

### 第三方应用商店 青龙优点
+ 持久化路径，数据安全无忧

卸载重装只要参数一致，可快速恢复至原状态，迁移只需要将数据目录复制到其他服务器原路径下，安装时填写相同参数即可无缝衔接

+ 更新及时
版本更新及时，预留至少3+的历史版本，安装时供用户选择

+ 支持网络模式选择
允许用户选择不同的网络模式，官方版暂不支持

## 升级 青龙面板
+ 点击升级
![image](https://github.com/whyour/qinglong/assets/56836951/54a34943-e50c-474e-a3c9-dd4662b5de8f)

+ 确认升级
![image](https://github.com/whyour/qinglong/assets/56836951/f9b3d589-942b-40b9-a74d-8c6c6e01a66b)

**Comment by donald001** [comment]
这个也得先安装Docker吧？我直接安装报错了：
```
[1Panel Log]: ======================= 开始安装 ======================= 
设置 1Panel 安装目录（默认为/opt）：
[1Panel Log]: 请输入目录的完整路径 
设置 1Panel 安装目录（默认为/opt）：/opt/1Panel
[1Panel Log]: 您选择的安装路径为 /opt/1Panel 
[1Panel Log]: ... 在线安装 docker 
选择延迟最低的源 https://mirrors.aliyun.com/docker-ce，延迟为 0.199528 秒
# Executing docker install script, commit: 6d9743e9656cc56f699a64800b098d5ea5a60020
+ sh -c apt-get update -qq >/dev/null
E: 仓库 “http://mirrors.tuna.tsinghua.edu.cn/AdoptOpenJDK/deb buster Release” 不再含有 Release 文件。
E: 仓库 “http://repo.mobian-project.org bullseye Release” 不再含有 Release 文件。
[1Panel Log]: ... 启动 docker 
Failed to enable unit: Unit file docker.service does not exist.
Failed to start docker.service: Unit docker.service not found.
[1Panel Log]: docker 安装失败 
```

**Reply by QYG2297248353** [reply]
请问您是什么发行版系统

**Comment by QYG2297248353** [comment]
【AlmaLinux OS 9】安装 1Panel 面板
发行版官网： [AlmaLinux OS 9](https://almalinux.org/zh-hans/get-almalinux/) 

+ 手动安装Docker
虽然 AlmaLinux 属于 RedHat/CentOS 系列的发行版，但由于1Panel 官方安装脚本不支持(Docker 安装脚本)，所以需要手动安装Docker.
```sh
yum install yum-utils
yum-config-manager --add-repo https://download.docker.com/linux/centos/docker-ce.repo
yum install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin
```

+ 安装1Panel
```sh
curl -sSL https://resource.fit2cloud.com/1panel/package/quick_start.sh -o quick_start.sh && sh quick_start.sh
```

**Reply by donald001** [reply]
不想安装docker，因为docker太TMD占内存了！

**Reply by QYG2297248353** [reply]
那么此教程不太适合你，2c2g服务器完全没有问题

**Reply by QYG2297248353** [reply]
Linux 系统下，dokcker几乎不占用内存，占用内存的实际是安装的容器，青龙面板下未运行任何脚本时，内存占用可以忽略不记就几十兆，实际还占用取决于您运行的脚本所需要的资源

windows docker 占用多那是没办法，WSL多少都是要占用固定内存大小的

**Comment by monsuns** [comment]
https://install.lifebus.top/auto_app_install.sh 已经404了. 是停止维护了还是换网址了?

**Reply by QYG2297248353** [reply]
## 【地址已变更】
### 自动化安装

一键式安装，无需手动操作，自动完成 1Panel应用 上架应用商店。

每三小时自动更新一次应用列表。

```sh
curl -sSL https://install.lifebus.top/auto_install.sh | bash
```

> 卸载自动化脚本 (不会卸载应用)

```sh
curl -sSL https://install.lifebus.top/auto_uninstall.sh | bash
```

如需卸载应用列表，您可以手动删除以下目录：

+ 应用目录：`${1panel应用目录}/resource/apps/local`

+ 应用公共文件目录：`/etc/1panel/envs`

+ 应用数据目录：`${应用持久化目录}`

### 手动安装模式

更新时机由自己把握，手动执行命令。

```sh
curl -sSL https://install.lifebus.top/app_install.sh | bash
```


请按照最新说明安装应用库：https://github.com/QYG2297248353/appstore-1panel
